ql/cashflows/overnightindexedcoupon.cpp

```diff
@@ -85,23 +95,25 @@ namespace QuantLib {
 
                 // forward part using telescopic property in order
                 // to avoid the evaluation of multiple forward fixings
-                if (i<n) {
-                    Handle<YieldTermStructure> curve =
-                        index->forwardingTermStructure();
+                if (date > today && valueDates[i] < endDate) {
```

Comment:
Same here.  If `date <= today`, then `i == n`, i.e., we got to the end of the dates in the fixed part.
You don't need to change this part of the code.